homeassistant/components/homekit/type_covers.py

```diff
@@ -253,14 +256,18 @@ def update_state(self, new_state):
                 or abs(current_position - self._homekit_target)
                 < DEVICE_PRECISION_LEEWAY
             ):
-                self.char_target_position.set_value(current_position)
+                if self.char_target_position.value != current_position:
+                    self.char_target_position.set_value(current_position)
                 self._homekit_target = None
         if new_state.state == STATE_OPENING:
-            self.char_position_state.set_value(1)
+            if self.char_position_state.value != 1:
+                self.char_position_state.set_value(1)
```

Comment:
Didn't you add constants for this in a previous PR?

Reply:
Different covers types, different values.

Reply:
Will add some more constants

Reply:
done.